dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -308,8 +308,12 @@ def select_level
     end
 
     # Check to see if any of the variants are part of an experiment that we're in
-    experiment_level = @script_level.find_experiment_level(current_user, @section)
-    return experiment_level if experiment_level
+    if current_user
+      section_as_student = current_user.sections_as_student.find_by(script: @script_level.script) ||
+        current_user.sections_as_student.first
+      experiment_level = @script_level.find_experiment_level(current_user, section_as_student)
+      return experiment_level if experiment_level
+    end
```

Comment:
this seems like a potentially expensive query. Is this being made every page load? Do we have a sense of how much this is adding to the page load ... load?